### PR TITLE
Implements Analyze Client

### PR DESCRIPTION
**Pull request by dvonthenen**
Addresses Issues: https://github.com/deepgram/deepgram-dotnet-sdk/issues/158, https://github.com/deepgram/deepgram-dotnet-sdk/issues/157, https://github.com/deepgram/deepgram-dotnet-sdk/issues/156, https://github.com/deepgram/deepgram-dotnet-sdk/issues/155

This implements Analyze Client using PreRecorded Client as a template. This should be working... just need the unit tests and examples to make sure they do.

TODOs:
will add unit tests and examples in subsequent PRs

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `ReadClient` class for interacting with Read API's, including transcription from URLs, byte arrays, streams, and support for callback URLs for asynchronous tasks.
	- Added `AnalyzeSchema` class with extensive properties for transcription customization.
	- Introduced `AsyncResponse`, `Average`, `Intent`, `IntentGroup`, `IntentsInfo`, `Metadata`, `Results`, `Segment`, `SentimentGroup`, `SentimentInfo`, `Summary`, `SummaryInfo`, `SyncResponse`, `Topic`, `TopicGroup`, `TopicsInfo`, and `UrlSource` classes in the `Analyze` module.
	- Enhanced `Live` and `Manage` modules with new records and schema updates for better data handling and response structures.

- **Refactor**
	- Updated import statements across various modules for improved code management and version control.
	- Transitioned several classes to records for immutable data structures in both `Live` and `PreRecorded` modules.

- **Bug Fixes**
	- Corrected missing semicolons and syntax errors in various `Manage` module classes to ensure code integrity and proper compilation.

### CODE REVIEW

**Review by CopperBeardy — DISMISSED** [review]

**Comment by dvonthenen** [comment]
@coderabbitai review

File: Deepgram/Models/Analyze/v1/TopicGroup.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Deepgram.Models.Analyze.v1;
+
+public record TopicGroup
+{
+    /// <summary>
+    /// TODO
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Analyze/v1/TopicGroup.cs:6]: will come back and do TODOs in a subsequent PR

File: Deepgram/Models/Analyze/v1/SentimentInfo.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿namespace Deepgram.Models.Analyze.v1;
+
+public class SentimentInfo
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Analyze/v1/SentimentInfo.cs:3]: will fix

File: Deepgram/Models/Analyze/v1/SentimentGroup.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+﻿namespace Deepgram.Models.Analyze.v1;
+
+public class SentimentGroup
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Analyze/v1/SentimentGroup.cs:3]: will update

File: Deepgram/Models/Analyze/v1/Segment.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
+﻿namespace Deepgram.Models.Analyze.v1;
+
+public class Segment
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Analyze/v1/Segment.cs:3]: will update

**Review by SandraRodgers — APPROVED** [review]

**Pull request merged**